### DISCUSSION TITLE
Remove machines before upgrading to 5.0?

### DISCUSSION THREAD

**Discussion by carrete** [containers/podman#22114] [Q&A]
I just ran `brew update`, and podman was upgraded to 5.0. After the upgrade podman said no machines were detected, even though there was one (used yesterday). I ran `podman machine reset`, but I'm not sure it removed the previously created machine. Is it necessary to delete machines created with v4 before upgrading to v5? How can I check that the machine really was removed?

**Comment by tnk4on** [comment]
Running `podman machine rm` in v4 will delete the following files (for QEMU).
```
% podman machine rm

The following files will be deleted:

/Users/user/.ssh/podman-machine-default
/Users/user/.ssh/podman-machine-default.pub
/Users/user/.config/containers/podman/machine/qemu/podman-machine-default.ign
/Users/user/.local/share/containers/podman/machine/qemu/podman-machine-default_fedora-coreos-39.20240309.2.0-qemu.aarch64.qcow2
/Users/user/.local/share/containers/podman/machine/qemu/podman.sock
/Users/user/.local/share/containers/podman/machine/qemu/podman-machine-default_ovmf_vars.fd
/Users/user/.config/containers/podman/machine/qemu/podman-machine-default.json
```

Running `podman machine reset` in v5 deleted the files except for the .ssh key.
```
% ls -ld /Users/user/.ssh/podman-machine-default
ls -ld /Users/user/.ssh/podman-machine-default.pub
ls -ld /Users/user/.config/containers/podman/machine/qemu/podman-machine-default.ign
ls -ld /Users/user/.local/share/containers/podman/machine/qemu/podman-machine-default_fedora-coreos-39.20240309.2.0-qemu.aarch64.qcow2
ls -ld /Users/user/.local/share/containers/podman/machine/qemu/podman.sock
ls -ld /Users/user/.local/share/containers/podman/machine/qemu/podman-machine-default_ovmf_vars.fd
ls -ld /Users/user/.config/containers/podman/machine/qemu/podman-machine-default.json
-rw-------@ 1 user  staff  411  3 21 22:01 /Users/user/.ssh/podman-machine-default
-rw-r--r--@ 1 user  staff  97  3 21 22:01 /Users/user/.ssh/podman-machine-default.pub
ls: /Users/user/.config/containers/podman/machine/qemu/podman-machine-default.ign: No such file or directory
ls: /Users/user/.local/share/containers/podman/machine/qemu/podman-machine-default_fedora-coreos-39.20240309.2.0-qemu.aarch64.qcow2: No such file or directory
ls: /Users/user/.local/share/containers/podman/machine/qemu/podman.sock: No such file or directory
ls: /Users/user/.local/share/containers/podman/machine/qemu/podman-machine-default_ovmf_vars.fd: No such file or directory
ls: /Users/user/.config/containers/podman/machine/qemu/podman-machine-default.json: No such file or directory
```

Please check the existence of qemu directories under ~/.config and ~/.local.

**Reply by carrete** [reply]
> Please check the existence of qemu directories under ~/.config and ~/.local.

Empty

**Comment by Luap99** [comment]
https://blog.podman.io/2024/03/migration-of-podman-4-to-podman-5-machines/

**Reply by carrete** [reply]
> Before upgrading to Podman 5, remove all of your Podman 4 machines.

👌 Perhaps the upgrade should abort if any v4 machines are detected?

**Comment by baude** [comment]
if you upgrade from podman 4 to 5 on mac, `podman machine reset` will not help you.  a manual cleanup is probably required.

this might help:

pre-checklist: if your machine was running when you upgraded, then look for processes like qemu and gvproxy and kill them.

1. look inside `~/.config/share/containers/podman/machine/qemu/` and note any machines (they would be `name.json`.  you can ignore `.ign` and `.lock` files.
2. rm those json files
3. now run `podman system connection ls` and note any connections with vm names
4. remove those connections with `podman system connection rm`
5. optional, remove any leftover ssh keys in ~/.ssh
6. now remove `~/.config/share/containers/podman/machine/qemu/` and `~/.local/share/containers/podman/machine/qemu`

that should get you close.
